interp/builtin.go

```diff
@@ -214,6 +214,12 @@ func (r *Runner) builtinCode(ctx context.Context, pos syntax.Pos, name string, a
 			path = r.envGet("HOME")
 		case 1:
 			path = args[0]
+
+			// replicate the commonly implemented behavior of `cd -`
+			// ref: https://www.man7.org/linux/man-pages/man1/cd.1p.html#OPERANDS
+			if len(path) == 1 && path[0] == '-' {
```

Comment:
```suggestion
			if path == "-" {
```

Reply:
ah yeah, maybe I was too skeptical on string comparison, I wrote in this verbose style to avoid accidental zero width characters in source code.

Anyway, I will accept your suggestion if you think that doesn't matter. ;)

Reply:
You mean zero width characters in the `"-"` string?

If that were a worry, I'd have to refactor hundreds of other lines :) Not a worry for now, so I'd say simplify.